### ISSUE TITLE
inlay hints for merged error sets show the entire body of the error type

### ISSUE THREAD

**Issue by xdBronch** [zigtools/zls#1828]
### Zig Version

0.12.0-dev.3245+4f782d1e8

### Zig Language Server Version

0.12.0-dev.490+fd3b5af

### Steps to Reproduce and Observed Behavior

```zig
test {
    const not_merged = @as(error{ a, b, c, d, e, f, g, h }, undefined);
    const merged = @as(error{ a, b, c, d } || error{ e, f, g, h }, undefined);

    _ = not_merged;
    _ = merged;
}
```
![image](https://github.com/zigtools/zls/assets/51252236/88001aa6-649e-4a5d-bdfa-0dbc87f9ab36)

the problem becomes much worse with many error types in the stdlib due to doc comments on each error

### Expected Behavior

error set should be truncated similar to what was done in #1822

**Comment by jkmdn-dev** [comment]
Anyone working on this? Mind if I take a stab at it if not?

**Comment by Techatrix** [comment]
Fixed by #1967
